core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java

```diff
@@ -103,16 +105,20 @@ public ExecutionDAOFacade(
                                     executor);
                             Monitors.recordDiscardedIndexingCount("delayQueue");
                         });
-
-        try {
-            Executors.newSingleThreadScheduledExecutor()
-                    .scheduleAtFixedRate(
-                            this::pruneWorkflowsAndTasks,
-                            0,
-                            pruningIntervalInMinutes(),
-                            TimeUnit.MINUTES);
-        } catch (Exception e) {
-            LOGGER.error("Error during pruning of workflows and tasks", e);
+        if ("true".equals(System.getProperty(CONDUCTOR_INDEXING_ENABLED))) {
```

Comment:
This can be something like
if (Boolean.parseBoolean(System.getProperty(CONDUCTOR_INDEXING_ENABLED))) {
}

Reply:
the properties file has string key value pair. This string check should be fast when compared to converting to boolean and then check for it.